vm/state_tree/src/lib.rs

```diff
@@ -335,33 +333,34 @@ where
     }
 
     /// Mutate and set actor state for an Address.
-    pub fn mutate_actor<F>(&mut self, addr: &Address, mutate: F) -> Result<(), Box<dyn StdError>>
+    pub fn mutate_actor<F>(&mut self, addr: &Address, mutate: F) -> Result<(), anyhow::Error>
     where
         F: FnOnce(&mut ActorState) -> Result<(), String>,
     {
         // Retrieve actor state from address
-        let mut act: ActorState = self
-            .get_actor(addr)?
-            .ok_or(format!("Actor for address: {} does not exist", addr))?;
+        let mut act: ActorState = self.get_actor(addr)?.ok_or(anyhow::anyhow!(
+            "Actor for address: {} does not exist",
+            addr
+        ))?;
 
         // Apply function of actor state
-        mutate(&mut act)?;
+        mutate(&mut act).map_err(|e| anyhow::anyhow!("{}", e))?;
         // Set the actor
         self.set_actor(addr, act)
     }
```

Comment:
Hmm, I wonder how this function is used. Maybe we could change the callback to be `F: FnOnce(&mut ActorState) -> Result<(), anyhow::Error>,`

Reply:
Done.